ember-core/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -133,49 +133,48 @@ private[ember] object Parser {
       var start = initStart
 
       while (!complete && idx < bv.size) {
-        (state: @switch) match {
```

Comment:
switch statements with fewer than 4 cases are never optimised, so we might as well do a boolean and an if